cli/lib/install.go

```diff
@@ -40,65 +40,92 @@ func initInstallCommand() *cobra.Command {
 		Run:  runInstallCommand,
 	}
 	installCommand.Flags().BoolVar(&installFlags.noDeps, "no-deps", false, "Do not install dependencies.")
+	installCommand.Flags().StringVarP(&installFlags.gitURL, "git-url", "g", "", "Enter git url for libraries hosted on repositories")
+	installCommand.Flags().StringVarP(&installFlags.zipPath, "zip-path", "z", "", "Enter a path to zip file")
 	return installCommand
 }
 
 var installFlags struct {
-	noDeps bool
+	noDeps  bool
+	gitURL  string
+	zipPath string
 }
 
 func runInstallCommand(cmd *cobra.Command, args []string) {
 	instance := instance.CreateInstanceIgnorePlatformIndexErrors()
-	libRefs, err := ParseLibraryReferenceArgsAndAdjustCase(instance, args)
-	if err != nil {
-		feedback.Errorf("Arguments error: %v", err)
-		os.Exit(errorcodes.ErrBadArgument)
-	}
-
-	toInstall := map[string]*rpc.LibraryDependencyStatus{}
-	if installFlags.noDeps {
-		for _, libRef := range libRefs {
-			toInstall[libRef.Name] = &rpc.LibraryDependencyStatus{
-				Name:            libRef.Name,
-				VersionRequired: libRef.Version,
-			}
+	if installFlags.zipPath != "" {
+		ZiplibraryInstallReq := &rpc.ZipLibraryInstallReq{
+			Instance: instance,
+			Path:     installFlags.zipPath,
+		}
+		_, err := lib.ZipLibraryInstall(context.Background(), ZiplibraryInstallReq)
+		if err != nil {
+			feedback.Errorf("Error installing Zip Library %v", err)
+			os.Exit(errorcodes.ErrGeneric)
+		}
+	} else if installFlags.gitURL != "" {
+		GitlibraryInstallReq := &rpc.GitLibraryInstallReq{
+			Instance: instance,
+			Name:     args[0],
```

Comment:
Why `args[0]` as `Name`? Can extract it directly from the `gitURL` inside the `GitLibraryInstall` function?

Reply:
So I thought it'd be better if we followed the same syntax of `arduino-cli lib install LIBRARY[@VERSION_NUMBER](S) [flags]`. Even though it seems redundant, I thought it would minimize confusion as the same command would have different arguments for different flags.

Reply:
Ah! I see why you did it but I think it would be better to avoid the redundancy, also if we go that way we'd have to do it for the installation suing the zip file too.